internal/trie/node/decode.go

```diff
@@ -36,42 +34,37 @@ func Decode(reader io.Reader) (n Node, err error) {
 	}
 	header := oneByteBuf[0]
 
-	nodeType := Type(header >> 6)
-	switch nodeType {
-	case LeafType:
+	nodeTypeHeaderByte := header >> 6
+	switch nodeTypeHeaderByte {
+	case leafHeader:
 		n, err = decodeLeaf(reader, header)
 		if err != nil {
 			return nil, fmt.Errorf("cannot decode leaf: %w", err)
 		}
 		return n, nil
-	case BranchType, BranchWithValueType:
+	case branchHeader, branchWithValueHeader:
 		n, err = decodeBranch(reader, header)
 		if err != nil {
 			return nil, fmt.Errorf("cannot decode branch: %w", err)
 		}
 		return n, nil
 	default:
-		return nil, fmt.Errorf("%w: %d", ErrUnknownNodeType, nodeType)
+		return nil, fmt.Errorf("%w: %d", ErrUnknownNodeType, nodeTypeHeaderByte)
 	}
 }
 
-// decodeBranch reads and decodes from a reader with the encoding specified in lib/trie/node/encode_doc.go.
+// decodeBranch reads and decodes from a reader with the encoding specified in internal/trie/node/encode_doc.go.
 // Note that since the encoded branch stores the hash of the children nodes, we are not
 // reconstructing the child nodes from the encoding. This function instead stubs where the
 // children are known to be with an empty leaf. The children nodes hashes are then used to
 // find other values using the persistent database.
-func decodeBranch(reader io.Reader, header byte) (branch *Branch, err error) {
-	nodeType := Type(header >> 6)
-	switch nodeType {
-	case BranchType, BranchWithValueType:
-	default:
-		return nil, fmt.Errorf("%w: %d", ErrNodeTypeIsNotABranch, nodeType)
```

Comment:
I understand that `decodeBranch` gets called only on a branch, so we could do without this check. 
By removing this check we rely on the caller to not use `decodeBranch` other than for a branch.
I think having this check does not hurt.